src/identity-utils.ts

```diff
@@ -0,0 +1,160 @@
+): string => {
+    // set DAS first since it is not an official identity type
+    let cacheKey: string = `${method}:device_application_stamp=${userIdentities.device_application_stamp};`;
+    const idLength: number = Object.keys(userIdentities).length;
+    let concatenatedIdentities: string = '';
+
+    if (idLength) {
+        let userIDArray: Array<string> = new Array(idLength);
+        // create an array where each index is equal to the user identity type
+        for (let key in userIdentities) {
+            if (key === 'device_application_stamp') {
+                continue;
+            } else {
+                userIDArray[Types.IdentityType.getIdentityType(key)] =
+                    userIdentities[key];
+            }
+        }
+
+        concatenatedIdentities = userIDArray.reduce(
+            (prevValue: string, currentValue: string, index: number) => {
+                let idName: string = Types.IdentityType.getIdentityName(index);
+                return `${prevValue}${idName}=${currentValue};`;
+            },
+            cacheKey
+        );
+    }
+
+    return concatenatedIdentities;
+};
+
+export const hasValidCachedIdentity = (
+    method: string,
+    proposedUserIdentities: IKnownIdentities,
+    idCache?: LocalStorageVault<Dictionary>
+): Boolean => {
+    // There is an edhge case where multiple identity calls are taking place 
+    // before identify fires, so there may not be a cache.  See what happens when 
+    // the ? in idCache is removed to the following test
+    // "queued events contain login mpid instead of identify mpid when calling 
+    // login immediately after mParticle initializes"
+    const cache = idCache?.retrieve();
+
+    // if there is no cache, then there is no valid cached identity
+    if (!cache) {
+        return false;
+    }
+
+    const cacheKey: string = concatenateIdentities(
+        method,
+        proposedUserIdentities
+    );
+
+    // if cache doesn't have the cacheKey, there is no valid cached identity
+    if (!cache.hasOwnProperty(cacheKey)) {
+        return false;
+    }
+    
+    // If there is a valid cache key, compare the expireTimestamp to the current time
+    // and return 
+    const expireTimestamp = cache[cacheKey].expireTimestamp;
+    
+    // if the current time is greater than the expireTimestamp, it is not a valid
+    // cached identity.
+    if (expireTimestamp < new Date().getTime()) {
+        return false;
+    } else {
+        return true;
+    }
+};
+
+export const getCachedIdentity = (
+    method: string,
+    proposedUserIdentities: IKnownIdentities,
+    idCache: LocalStorageVault<Dictionary>
+): Dictionary<string | number | boolean> | null => {
+    const cacheKey: string = concatenateIdentities(
+        method,
+        proposedUserIdentities
+    );
+
+    const cache = idCache.retrieve();
+    const cachedIdentity = cache ? cache[cacheKey] : null;
+
+    return cachedIdentity;
+};
+
+export const createKnownIdentities = (
```

Comment:
create a new file to put non-cache utility methods in

Reply:
I think this can actually be done in a future ticket when identity.js gets refactored into identity.ts.  Right now this is the only non-cache related utility method.